src/main/java/models/Book.java

```diff
@@ -80,7 +81,10 @@ public void removeShoppingCarts(){
     public String getPublisher() { return this.publisher; }
     public void setPublisher(String publisher) { this.publisher = publisher; }
 
-    public boolean equals(Book book){
+    @Override
+    public boolean equals(Object object){
```

Comment:
is this just to be safe? or is there a reason this guard is here?

Reply:
this overrides the object Equals method, which  is used in arraylists and the such